merge-build-push/merge-and-build.sh

```diff
@@ -15,7 +18,10 @@ OSE_VERSION="${MAJOR}.${MINOR}"
 if [ "${OSE_VERSION}" != "${OSE_MASTER}" ] ; then
   PUSH_EXTRA="--nolatest"
 fi
-BUILDPATH="${HOME}/go"
+
+# Use the directory relative to this Jenkins job.
+BUILDPATH="${WORKSPACE}/go"
+mkdir -p $BUILDPATH
 cd $BUILDPATH
 export GOPATH=`pwd`
```

Comment:
Also don't use back-ticks -- they're a deprecated feature. Use the `$()` subshell expansion.

And don't mix scoping statements (`export`, `readonly`, `local`, `declare`) with something else that has a return code on the same line.

```sh
GOPATH="$( pwd )"; export GOPATH
```